### ISSUE TITLE
Missing details about "Build.Cleanup" capability that can be added to agents to route clean up jobs

### ISSUE THREAD

**Issue by yaananth** [MicrosoftDocs/vsts-docs#3685]
Build.Cleanup capability can be used to direct clean up jobs only to certain agents, if we don't want clean up jobs hammering all agents. It's worth mentioning this in the doc.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 43a285f9-738d-9f76-4de0-a4c4ecf6f5f5
* Version Independent ID: 9581a043-4873-bf0a-138f-85f6a61ae950
* Content: [Build and release retention policies - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/policies/retention?view=azure-devops#feedback)
* Content Source: [docs/pipelines/policies/retention.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/pipelines/policies/retention.md)
* Product: **devops**
* GitHub Login: @andyjlewis
* Microsoft Alias: **alewis**

**Comment by moswald** [comment]
@andyjlewis This feature is controlled by a hidden feature flag in versions up to and including Azure DevOps Server 2019. I'll remove it from future versions, but send me a message if you've got questions.

**Comment by chasewilson** [comment]
@yaananth Thanks for the feedback. It's not a publicized feature and removed in future versions. If it is something you might like to see as a feature or a similar feature, you can reach out to the [Developer Community](https://developercommunity.visualstudio.com/spaces/21/index.html) if you want to put out a feature request. 
Feel free to open a new issue if you have any more questions.

**Comment by moswald** [comment]
@chasewilson I think you have that backward, maybe. The feature is live for everyone now, where it used to only be turned on by a feature flag upon request.

**Comment by chasewilson** [comment]
Whoops @moswald Thank you! I did misread it that's my fault I'll get it added in right away! Thanks for the quick catch!

**Comment by moswald** [comment]
@chasewilson Did you ever get this into the docs? I don't see it as part of `retention.md`, so maybe it was added somewhere else?

**Comment by japj** [comment]
I'm also interested in any information to this feature (using Azure DevOps Server 2019 Update1).
Any information on how this actually works/should be configured would be appreciated!

**Comment by moswald** [comment]
@japj To summarize for you and anyone else who may stumble upon this thread:

TL;DR: setting a `Build.Cleanup` capability on agents will cause the pool's cleanup jobs to be directed to just those agents, leaving the rest free to do regular work.

The gist is that when a pipeline run is deleted we need to clean up any associated artifacts. Artifacts that are stored outside of Azure DevOps/TFS need to be cleaned up through a job run on the agents. This is because the Azure DevOps service won't have permissions to delete those files, but the agent should be able to since it had the access needed to write the files originally. We attempt to queue the job on the deleted run's original agent pool, and at that point it's treated like any other job: it gets put into a queue and waits its turn.

Sometimes we clean up a lot of runs at once, and the agent pool gets saturated with cleanup jobs. This can cause "real work" like CI/CD runs to get backed up (in some cases, for a very long time). The solution to that is to designate a subset of agents in the pool that are the cleanup agents. If any agents have the `Build.Cleanup` capability set (the value can be anything, even blank, but I recommend using `true` or `1` in case we eventually check), we will make sure only those agents run the cleanup jobs, leaving the rest free to continue running your day-to-day pipeline jobs.

Unfortunately, @chasewilson is no longer our docs writer, and I'm not sure who should get assigned this task. @wnjenkin can you help out here?

**Comment by JaapMosselman** [comment]
@moswald I never see jobs running for cleanup local artifacts. As far as I can see in this documentation, I should not have to anything special. The pipeline runs are removed after the retention period, but the local artifacts are never removed. Any idea what the problem can be?
Also I see, that pipeline runs that ever where used by a release pipeline, seems to be retained for ever, even the releases are already deleted due to retension experation.

**Comment by moswald** [comment]
Hi @JaapMosselman can you file a new ticket on devcommunity.visualstudio.com for Azure DevOps? Someone (likely me) can help you figure this out. It's not a bug in the documentation so I'd rather move the discussion over there. Thanks!
